### PR TITLE
dev/core#27 Move check for phone function to a check

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#27 Move check for phone function to a check
https://lab.civicrm.org/dev/core/-/issues/27

Before
----------------------------------------
The check is run when crudding phone numbers

After
----------------------------------------
The check is part of system status checks

image

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Core/DAO.php [hunk 1]
```diff
@@ -2460,23 +2460,6 @@ public static function triggerRebuild($tableName = NULL, $force = FALSE) {
...
-  /**
-   * Because sql functions are sometimes lost, esp during db migration, we check here to avoid numerous support requests
-   * @see http://issues.civicrm.org/jira/browse/CRM-13822
-   * TODO: Alternative solutions might be
-   *  * Stop using functions and find another way to strip numeric characters from phones
-   *  * Give better error messages (currently a missing fn fatals with "unknown error")
-   */
-  public static function checkSqlFunctionsExist() {
-    if (!self::$_checkedSqlFunctionsExist) {
-      self::$_checkedSqlFunctionsExist = TRUE;
```
Comment (colemanw) [inline CRM/Core/DAO.php:2472]: Also you can delete this static variable from the class

**Comment by colemanw** [comment]
This is going to be a little less fool-proof than the original version which would guarantee the sql function exists before writing a phone record (also at the expense of an extra query before writing phone records).

But checks are a much more standard approach, than the ad-hoc thing.

File: CRM/Core/DAO.php (hunk 1 above)
Comment (colemanw) [reply CRM/Core/DAO.php:2472]: I fixed it.

**Comment by demeritcowboy** [comment]
All the PR test sites are showing this warning. Not sure yet if it's something to do with the way those sites are created since don't see it on e.g dmaster.demo.

**Pull request merged**